Update KumaScript {{anch}} macro calls

Vaent opened the pull request:
It looks like various section headers have been expanded on at some point without updating associated KumaScript, so broken links are being inserted.

Note: for completeness I have copied the current section headers as displayed in the HTML, however from the regexp in the [anch.ejs source code](https://github.com/mdn/yari/blob/main/kumascript/macros/anch.ejs) it looks as though brackets will cause problems if retained in the first argument, i.e. producing `#png_(portable_network_graphics)` instead of `#png_portable_network_graphics`. In this case another pass is needed to remove all brackets from section names passed into {{anch}} or else a tweak to the macro itself may be more generally useful.

Display text has been left unaltered in all calls and is now the second argument.

Macros other than {{anch}} have not been inspected.

If KumaScript is to be phased out anyway (as the [MDN guide](https://developer.mozilla.org/en-US/docs/MDN/Tools/KumaScript) implies) then hardcoded HTML/MD hyperlinks may be preferred to the changes proposed here.

<!-- 👀 Thanks for opening a PR! Read comments like this one to get your PR merged faster. -->
#### Summary
<!-- ✍️ In a sentence or two, describe your changes -->
Flagging KumaScript macro calls not working as intended. There are multiple routes to addressing the issue more or less fully as summarised above - further work required depending on preferred solution.

#### Motivation
<!-- ❓ Why are you making this change? Help us understand how your changes help readers. -->
Discovered broken links while browsing content.

#### Supporting details
<!-- 🔗 Link to supporting information, such as bug trackers, source control, release notes, and vendor docs. -->

#### Related issues
<!-- 🔨 If applicable, use "Fixes #XYZ" -->

#### Metadata
<!-- ✅ Check a box if applicable, like this: [x]

This PR…
-->
- [ ] Adds a new document
- [ ] Rewrites (or significantly expands) a document
- [x] Fixes a typo, bug, or other error

<!-- 👷‍♀️ After submitting, review the results of the "Checks" tab! -->

Vaent committed:
Update KumaScript {{anch}} macro calls

It looks like various section headers have been expanded on at some point without updating associated KumaScript, so broken links are being inserted.

Note: for completeness I have copied the current section headers as displayed in the HTML, however from the regexp in the [anch.ejs source code](https://github.com/mdn/yari/blob/main/kumascript/macros/anch.ejs) it looks as though brackets will cause problems if retained in the first argument, i.e. producing `#png_(portable_network_graphics)` instead of `#png_portable_network_graphics`. In this case another pass is needed to remove all brackets from section names passed into {{anch}} or else a tweak to the macro itself may be more generally useful.

Display text has been left unaltered in all calls and is now the second argument.

Macros other than {{anch}} have not been inspected.

If KumaScript is to be phased out anyway (as the [MDN guide](https://developer.mozilla.org/en-US/docs/MDN/Tools/KumaScript) implies) then hardcoded HTML/MD hyperlinks may be preferred to the changes proposed here.

wbamberg commented:
Thanks for your PR, @Vaent !

> It looks like various section headers have been expanded on at some point without updating associated KumaScript, so broken links are being inserted.
> 
> Note: for completeness I have copied the current section headers as displayed in the HTML, however from the regexp in the [anch.ejs source code](https://github.com/mdn/yari/blob/main/kumascript/macros/anch.ejs) it looks as though brackets will cause problems if retained in the first argument, i.e. producing `#png_(portable_network_graphics)` instead of `#png_portable_network_graphics`.

For some reason the preview URL wasn't generated for your PR, but I've checked out your branch locally and can confirm that yes, this is a problem and the `{{anch}}` macro does not generate correct fragments in this case.

> In this case another pass is needed to remove all brackets from section names passed into {{anch}} or else a tweak to the macro itself may be more generally useful.
> 
> Display text has been left unaltered in all calls and is now the second argument.
> 
> Macros other than {{anch}} have not been inspected.
> 
> If KumaScript is to be phased out anyway (as the [MDN guide](https://developer.mozilla.org/en-US/docs/MDN/Tools/KumaScript) implies) then hardcoded HTML/MD hyperlinks may be preferred to the changes proposed here.

In my opinion: yes, plain MD links would be better here. But I'll ask the other MDN maintainers to pitch in. I think if we do still want to use `{{anch}}` here, changing the argument to match the actual generated fragments would be better than updating the macro.

teoli2003 commented:
I think plain links are better here. 

Anchors are fragile, we should have a test that prevent merging a page if internal links are broken (doesn't cover the case where we link to an anchor on another page, but would be better than nothing).

wbamberg commented:
@Vaent , I've asked around and noone wants to keep `{{anch}}` so if you wanted to update this PR to remove it in this page (or even just in these places) I'd be very happy to merge it :).

wbamberg commented:
> I think plain links are better here.
> 
> Anchors are fragile, we should have a test that prevent merging a page if internal links are broken (doesn't cover the case where we link to an anchor on another page, but would be better than nothing).

-> https://github.com/mdn/yari/issues/5375

Vaent committed:
Merge branch 'mdn:main' into patch-1

Vaent committed:
Replace {{anch}} with MD links

wbamberg commented on files/en-us/web/media/formats/image_types/index.md:
```diff
@@ -415,7 +415,7 @@ The simplest, and most commonly used, form of BMP file is an uncompressed raster
 These features include: support for different bit depths, indexed color, alpha channels, and different pixel orders (by default, BMP is written from bottom-left corner toward the right and top, rather than from the top-left corner toward the right and bottom).
 
-Theoretically, several compression algorithms are supported, and the image data can also be stored in {{anch("JPEG")}} or {{anch("PNG")}} format within the BMP file.
+Theoretically, several compression algorithms are supported, and the image data can also be stored in [JPEG](#jpeg-joint-photographic-experts-group-image) or [PNG](#png-portable-network-graphics) format within the BMP file.
```
```suggestion
Theoretically, several compression algorithms are supported, and the image data can also be stored in [JPEG](#jpeg_joint_photographic_experts_group_image) or [PNG](#png_portable_network_graphics) format within the BMP file.
```

wbamberg commented on files/en-us/web/media/formats/image_types/index.md:
```diff
@@ -517,7 +517,7 @@ Theoretically, several compression algorithms are supported, and the image data
 In 1987, the CompuServe online service provider introduced the **{{interwiki("wikipedia", "GIF")}}** (**Graphics Interchange Format**) image file format to provide a compressed graphics format that all members of their service would be able to use.
 GIF uses the {{interwiki("wikipedia", "Lempel-Ziv-Welch")}} (LZW) algorithm to losslessly compress 8-bit indexed color graphics.
-GIF was one of the first two graphics formats supported by {{Glossary("HTML")}}, along with {{anch("XBM")}}.
+GIF was one of the first two graphics formats supported by {{Glossary("HTML")}}, along with [XBM](#xbm-x-window-system-bitmap-file).
```
```suggestion
GIF was one of the first two graphics formats supported by {{Glossary("HTML")}}, along with [XBM](#xbm_x_window_system_bitmap_file).
```

wbamberg reviewed (changes requested):
Thank you, @Vaent !

Some of the updates are using hyphens where they need to use underscores. I added suggestions for the first two, but they would all need to be updated.

And thank you also for starting off the conversation about `{{anch}}`, it seems like we should work towards retiring this macro :).

Vaent committed:
Fix typos in some new MD links

Vaent committed:
Insert reference-style links

This commit applies a pattern to reduce repetition and hopefully make maintenance easier.

For links to sections in the page, a reference has been added under the relevant header and all instances of the link replaced with the reference. If headers change in future, only the reference needs to be updated rather than every individual link. Having the reference adjacent to the header also acts as a prompt to make the necessary change.

References have only been created for headers which were identified as linked to within this page, when {{anch}} calls were replaced with MD links.

This commit is only a suggested approach and can be rolled back if the preference is to keep standard links throughout.

Vaent commented:
@wbamberg thanks for your attention, I spotted the underscores issue a bit too late and since I started out editing in GitHub I guess the PR got updated before I could clean up after myself! Although I had thought the underscores were incorrect and needed replacing with hyphens, so I need to double check whether I did indeed fix it or only made it worse...

Vaent commented:
I've now checked the preview in GitHub and looks like the hyphens are correct, underscores were incorrect so my most recently pushed commits should be fine. The last one is optional, a neater solution to my mind but depends on how you do things in MDN, if it is not suitable then commit 661c199 should do what's needed with nothing superfluous.

Vaent commented:
It took a fridge moment for me to realise I was making changes based on GitHub-flavoured MD which will not work with the MDN platform, I'll go back and change those pesky hyphens to underscores and scrap the references since I can't properly test whether or not they would work correctly.

wbamberg commented:
> It took a fridge moment for me to realise I was making changes based on GitHub-flavoured MD which will not work with the MDN platform, I'll go back and change those pesky hyphens to underscores and scrap the references since I can't properly test whether or not they would work correctly.

Yes, the fragment-generation is different for MDN compared with GitHub (idk why). The preview URL does seem to be working now though: https://pr13347.content.dev.mdn.mozit.cloud/en-US/docs/Web/Media/Formats/Image_types should show what the page would look like on MDN.

I don't think we should use reference-style links here though - I can see the appeal of them but we haven't in general used them in MDN Markdown yet, and it would be nice if we did want to do this to make a project to do it consistently.

Vaent committed:
Revert "Insert reference-style links"

This reverts commit a7f83fe1df76455a862b94fb701fb9a974deb848.

Vaent committed:
Replace hyphens with underscores in anchor links

wbamberg commented on files/en-us/web/media/formats/image_types/index.md:
```diff
@@ -1566,8 +1566,8 @@ If you do choose a lossy format, such as JPEG or lossy WebP, carefully weigh the
     <tr>
-      <td>{{anch("SVG")}}</td>
-      <td>{{anch("PNG")}}</td>
+      <td>[SVG](#svg_scalable_vector_graphics)</td>
+      <td>[PNG](#png_portable_network_graphics)</td>
```
```suggestion
      <td><a href="#svg_scalable_vector_graphics">SVG</a></td>
      <td><a href="#png_portable_network_graphics">PNG</a></td>
```

We don't process Markdown inside HTML, so this unfortunately has to be HTML. Or we could convert the whole table to Markdown: https://developer.mozilla.org/en-US/docs/MDN/Contribute/Markdown_in_MDN#tables but it might be better to keep it as HTML since all the other tables in this page are HTML.

Vaent replied:
Makes perfect sense, I've committed your suggestion and there don't appear to be any other instances like that

Vaent committed:
Replace Markdown tags in HTML table

Co-authored-by: wbamberg <will@bootbonnet.ca>

wbamberg reviewed (approved):
👍 OK, looks great. Thank you @Vaent !
